deeplake/core/tensor.py

```diff
@@ -350,9 +350,10 @@ def extend(
         Raises:
             TensorDtypeMismatchError: Dtype for array must be equal to or castable to this tensor's dtype.
         """
+        old_dataset_length = self.num_samples
```

Comment:
is this change really necessary?

Reply:
Yes, in order to keep track of the initial records.